wit/basic-consumer.wit

```diff
@@ -0,0 +1,7 @@
+/// A basic consumer, commonly used for load-balancing.
+default interface basic-consumer {
+	use pkg.types.{message, error, receiver}
+
+	/// `receive-message` satisfied pull-based messaging requirements
+	receive-message(r: receiver) -> result<message, error>
+}
```

Comment:
Since so many things can go wrong in distributed messaging, it's useful to have a 
  `receive-timeout` method that accepts a timeout parameter,
so the program can avoid hangs when things do go wrong.

There's a subsequent problem of how to configure the timeout value, but I assume configuration is out of scope for this interface. Even if the timeout is hard-coded, it's better than not having a timeout option.

Reply:
Strongly agree that we should have a receive-timeout method

Reply:
All methods that interact with the network need a timeout option.

Reply:
If a timeout is going to be added, would it make sense to have a structure closer to the Golang `context.Context`?

See: https://pkg.go.dev/context

Context would be useful for not just timeouts, but also for telemetry, trace propagation.